app/controllers/api_controller/settings.rb

```diff
@@ -0,0 +1,26 @@
+class ApiController
+  module Settings
+    def show_settings
+      validate_api_action
+      category = @req[:c_id]
+      selected_sections =
+        if category
+          raise NotFound, "Settings category #{category} not found" unless exposed_settings.include?(category)
+          category
+        else
+          exposed_settings
+        end
+
+      result = Array(selected_sections).each_with_object({}) do |section, hash|
```

Comment:
not directly, symbols vs. strings, but something like .slice(*Array(selected_sections).collect(&:to_sym)) would work. 

Thanks.